packages/react/src/components/AILabel/index.tsx

```diff
@@ -0,0 +1,365 @@
+} from '../Toggletip';
+import { IconButton } from '../IconButton';
+import { Undo } from '@carbon/icons-react';
+import { useId } from '../../internal/useId';
+import deprecate from '../../prop-types/deprecate';
+
+export type AILabelContentProps = React.HTMLAttributes<HTMLSpanElement>;
+
+export const AILabelContent = React.forwardRef(function AILabelContent(
+  { className, children, ...rest }: AILabelContentProps,
+  ref
+) {
+  const prefix = usePrefix();
+
+  const hasAILabelActions = React.Children.toArray(children).some((child) => {
+    const item = child as any;
+    item.type?.displayName === 'AILabelActions';
+  });
+
+  const aiLabelContentClasses = cx(className, {
+    [`${prefix}--slug-content`]: true,
+    [`${prefix}--slug-content--with-actions`]: hasAILabelActions,
+  });
+
+  return (
+    <ToggletipContent className={aiLabelContentClasses}>
+      {children}
+    </ToggletipContent>
+  );
+});
+
+AILabelContent.displayName = 'AILabelContent';
+AILabelContent.propTypes = {
+  /**
+   * Specify the content you want rendered inside the slug ToggleTip
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be added to the AI slug callout
+   */
+  className: PropTypes.string,
+};
+
+export type AILabelActionsProps = React.HTMLAttributes<HTMLSpanElement>;
+
+export const AILabelActions = React.forwardRef(function AILabelActions(
+  { className, children, ...rest }: AILabelActionsProps,
+  ref
+) {
+  const prefix = usePrefix();
+
+  const aiLabelActionsClasses = cx(className, {
+    [`${prefix}--slug-actions`]: true,
+  });
+
+  return (
+    <ToggletipActions className={aiLabelActionsClasses}>
+      {children}
+    </ToggletipActions>
+  );
+});
+
+AILabelActions.displayName = 'AILabelActions';
+AILabelActions.propTypes = {
+  /**
+   * Specify the content you want rendered inside the slug callout toolbar
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be added to the AI slug toolbar
+   */
+  className: PropTypes.string,
+};
+
+/**
+ * Deprecated popover alignment values.
+ * @deprecated Use NewPopoverAlignment instead.
+ */
+export type DeprecatedAlignment =
+  | 'top-left'
+  | 'top-right'
+  | 'bottom-left'
+  | 'bottom-right'
+  | 'left-bottom'
+  | 'left-top'
+  | 'right-bottom'
+  | 'right-top';
+
+export type NewAlignment =
+  | 'top'
+  | 'bottom'
+  | 'left'
+  | 'right'
+  | 'top-start'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'left-end'
+  | 'left-start'
+  | 'right-end'
+  | 'right-start';
+
+export type Alignment = DeprecatedAlignment | NewAlignment;
+
+const propMappingFunction = (deprecatedValue) => {
+  const mapping = {
+    'top-left': 'top-start',
+    'top-right': 'top-end',
+    'bottom-left': 'bottom-start',
+    'bottom-right': 'bottom-end',
+    'left-bottom': 'left-end',
+    'left-top': 'left-start',
+    'right-bottom': 'right-end',
+    'right-top': 'right-start',
+  };
+  return mapping[deprecatedValue];
+};
```

Comment:
Not a blocker to merge at all, but at some point we should probably make this a reusable helper - we're going to be using this all over the codebase 😅

Reply:
I totally forgot we have one already: 

https://github.com/carbon-design-system/carbon/blob/afce28df13ae3475eb32c7b65e0599eb86775458/packages/react/src/tools/createPropAdapter.js#L44-L67

Reply:
ooh thanks! I updated AI label and Popover to use this